spec/models/miq_request_spec.rb

```diff
@@ -412,4 +412,18 @@
       expect { described_class.class_from_request_data({}) }.to raise_error("Invalid request_type")
     end
   end
+
+  context "user with multiple groups" do
+    let(:root_tenant) { Tenant.seed }
+    let(:tenant1)     { FactoryGirl.create(:tenant, :parent => root_tenant) }
+    let(:group1)      { FactoryGirl.create(:miq_group, :description => 'Group 1', :tenant => root_tenant) }
+    let(:group2)      { FactoryGirl.create(:miq_group, :description => 'Group 2', :tenant => tenant1) }
+    let(:user)        { FactoryGirl.create(:user, :miq_groups => [group1, group2], :current_group => group1) }
+    let(:request)     { FactoryGirl.create(:miq_provision_request, :requester => user, :options => {:requester_group => group2.description}) }
+
+    it "stays with the requester group" do
+      expect(user.current_group).to eq(group1)
+      expect(request.get_user.current_group).to eq(group2)
+    end
```

Comment:
@lfu Can you also add a tests for the case where the user does not share a group with the `options[:requester_group]` value.  In  that case the user remains in their current group.